Source/ACE/Entity/WorldObject.cs

```diff
@@ -694,6 +695,433 @@ public void Examine(Session examiner)
+                                                         || x.PropertyId == (uint)PropertyDouble.ArmorModVsNether).ToList();
+            if (propertiesArmor.Count > 0)
+            {
+                flags |= IdentifyResponseFlags.ArmorProfile;
+            }
+
+            // Weapons Profile
+            var propertiesWeaponsD = PropertiesDouble.Where(x => x.PropertyId < 9000
+                                                            && (x.PropertyId == (uint)PropertyDouble.WeaponLength
+                                                            || x.PropertyId == (uint)PropertyDouble.DamageVariance
+                                                            || x.PropertyId == (uint)PropertyDouble.MaximumVelocity
+                                                            || x.PropertyId == (uint)PropertyDouble.WeaponOffense
+                                                            || x.PropertyId == (uint)PropertyDouble.DamageMod)).ToList();
+
+            var propertiesWeaponsI = PropertiesInt.Where(x => x.PropertyId < 9000
+                                                         && (x.PropertyId == (uint)PropertyInt.Damage
+                                                         || x.PropertyId == (uint)PropertyInt.DamageType
+                                                         || x.PropertyId == (uint)PropertyInt.WeaponSkill
+                                                         || x.PropertyId == (uint)PropertyInt.WeaponTime)).ToList();
+
+            if (propertiesWeaponsI.Count + propertiesWeaponsD.Count > 0)
+            {
+                flags |= IdentifyResponseFlags.WeaponProfile;
+            }
+
+            if (propertiesString.Count > 0)
+            {
+                flags |= IdentifyResponseFlags.StringStatsTable;
+            }
+
+            if (ItemType == ItemType.Creature)
+            {
+                flags |= IdentifyResponseFlags.CreatureProfile;
+            }
+
+            // Ok Down to business - let's identify all of this stuff.
+            WriteIdentifyObjectHeader(writer, flags, successfulId);
+            WriteIdentifyObjectIntProperties(writer, flags, propertiesInt);
+            WriteIdentifyObjectInt64Properties(writer, flags, propertiesInt64);
+            WriteIdentifyObjectBoolProperties(writer, flags, propertiesBool);
+            WriteIdentifyObjectDoubleProperties(writer, flags, propertiesDouble);
+            WriteIdentifyObjectStringsProperties(writer, flags, propertiesString);
+            WriteIdentifyObjectDidProperties(writer, flags, propertiesDid);
+            WriteIdentifyObjectSpellIdProperties(writer, flags, propertiesSpellId);
+            WriteIdentifyObjectArmorProfile(writer, flags, propertiesArmor);
+            // TODO: There are probably other checks that need to be made here
+            if (ItemType == ItemType.Creature && GetType().Name != "DebugObject")
+            {
+                WriteIdentifyObjectCreatureProfile(writer, (Creature)this);
```

Comment:
This should be handled in the "Creature" class via virtual method override, not here, as it is a creature exclusive profile.